### PR TITLE
Report progress for file sync

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#46

### CODE REVIEW

File: sparse/client.go [hunk 1]
```diff
@@ -285,6 +293,7 @@ func (client *syncClient) syncDataInterval(file FileIoProcessor, dataInterval In
 			}
 		}
 		offset += batchInterval.Len()
+		ops.UpdateSyncFileProgress(size)
```
Comment (yasker) [inline sparse/client.go]: Size? Are we going to use the accumulated result?

Comment (shuo-wu) [reply sparse/client.go]: Yes.

File: sparse/rest/handlers.go [hunk 2]
```diff
@@ -89,7 +99,10 @@ func (server *SyncServer) close(writer http.ResponseWriter, request *http.Reques
 
 	server.fileIo.Close()
 	log.Infof("Ssync server exit(0)")
```
Comment (shuo-wu) [inline sparse/rest/handlers.go]: modify this log

File: sparse/rest/handlers.go [hunk 3]
```diff
@@ -89,7 +99,10 @@ func (server *SyncServer) close(writer http.ResponseWriter, request *http.Reques
...
-	os.Exit(0)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	server.srv.Shutdown(ctx)
```
Comment (shuo-wu) [inline sparse/rest/handlers.go]: directly use `Close()`

**Pull request merged**